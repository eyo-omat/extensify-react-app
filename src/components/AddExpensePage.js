import React from 'react';
import ExpenseForm from './ExpenseForm';
import {connect} from 'react-redux';
import {startExpense} from "../actions/expenses";

export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startExpense(expense);
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                <h1>Add Expense</h1>
                <ExpenseForm
                    onSubmit={this.onSubmit}
                />
            </div>
        );
    }
}
const mapDispatchToProps = (dispatch) => ({startExpense: (expense) => dispatch(startExpense(expense))});


export default connect(undefined, mapDispatchToProps)(AddExpensePage);
