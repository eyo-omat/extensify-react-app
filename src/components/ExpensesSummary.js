import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpenseTotal from '../selectors/expenses-total';

const ExpensesSummary = ({expensesCount, expensesTotal}) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formatExpenseTotal = numeral(expensesTotal / 100).format('$0,0.00');

    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">
                    Viewing <span>{expensesCount}</span> {expenseWord} totalling <span>{formatExpenseTotal}</span>
                </h1>
                <div className="page-header__actions">
                    <Link to="/create" className="button">Add Expense </Link>

                </div>
            </div>
        </div>

    );
};

const mapStateToProps = (state) => {
    const visibleExpense = selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: visibleExpense.length,
        expensesTotal: selectExpenseTotal(visibleExpense)

    };

};

export default connect(mapStateToProps)(ExpensesSummary);


