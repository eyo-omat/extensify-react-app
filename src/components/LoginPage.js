import React from 'react';
import {startLogin} from '../actions/auth';
import { connect } from 'react-redux';

export const LoginPage = ({ startLogin }) => {

    return (
        <div className="box-layout">

            <div className="box-layout__box">
                <h1 className="box-layout__title">Expensify</h1>
                <p>Its time to get you expenses under control.</p>

                <button onClick={startLogin} className="button">Login with Google</button>
            </div>
        </div>

    );
};

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage)


