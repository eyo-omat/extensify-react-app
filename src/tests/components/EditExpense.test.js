import React from 'react';
import {shallow} from 'enzyme';
import {EditExpensePage} from '../../components/EditExpensePage';
import expenseData from '../fixtures/expenses';

let editExpense, removeExpense, history, wrapper;

beforeEach(() => {
    editExpense = jest.fn();
    removeExpense = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(<EditExpensePage editExpense={editExpense} removeExpense={removeExpense} expense={expenseData[1]} history={history} />)
});

test("should render Edit Expense page correctly", () => {
    expect(wrapper).toMatchSnapshot();

});

test("should handle editExpense", () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenseData[1]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(editExpense).toHaveBeenLastCalledWith(expenseData[1].id, expenseData[1]);


});

test("should handle removeExpense", () => {
    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(removeExpense).toHaveBeenLastCalledWith({id: expenseData[1].id});


});



