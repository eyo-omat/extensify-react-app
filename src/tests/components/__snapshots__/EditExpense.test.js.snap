// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Edit Expense page correctly 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 450,
        "createdAt": 345600000,
        "description": "Mac",
        "id": "2",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
